src/utilities/valueToLiteral.js

```diff
@@ -0,0 +1,97 @@
+import { inspect } from '../jsutils/inspect';
+import { isIterableObject } from '../jsutils/isIterableObject';
+
+import type { ConstValueNode } from '../language/ast';
+import { Kind } from '../language/kinds';
+
+import type { GraphQLInputType } from '../type/definition';
+import {
+  getNamedType,
+  isLeafType,
+  isInputObjectType,
+} from '../type/definition';
+
+/**
+ * Produces a GraphQL Value AST given a JavaScript value.
+ *
+ * A GraphQL type may be provided, which will be used to interpret different
+ * JavaScript values if it defines a `valueToLiteral` method.
+ *
+ * | JavaScript Value  | GraphQL Value        |
+ * | ----------------- | -------------------- |
+ * | Object            | Input Object         |
+ * | Array             | List                 |
+ * | Boolean           | Boolean              |
+ * | String            | String               |
+ * | Number            | Int / Float          |
+ * | null / undefined  | Null                 |
+ *
+ * Note: This function does not perform any type validation or coercion.
+ */
+export function valueToLiteral(
+  value: mixed,
+  type?: GraphQLInputType,
+): ConstValueNode {
+  // Like JSON, a null literal is produced for null and undefined.
+  if (value == null) {
+    return { kind: Kind.NULL };
+  }
+
+  const namedType = type && getNamedType(type);
+
+  // Convert JavaScript array to GraphQL list.
+  if (isIterableObject(value)) {
+    return {
+      kind: Kind.LIST,
+      values: Array.from(value, (item) => valueToLiteral(item, namedType)),
+    };
+  }
+
+  // Does this type (if provided) define `valueToLiteral` which returns a value?
+  if (isLeafType(namedType) && namedType.valueToLiteral != null) {
+    const literal = namedType.valueToLiteral(value);
+    if (literal) {
+      return literal;
+    }
+  }
+
+  // Otherwise, perform a JS-to-Literal default equivalency.
```

Comment:
we should make clear that this is deprecated, possibility wrong behaviour and Scalars should define their appropriate logic.

Reply:
This is not deprecated, these are the default conversions. Other languages might implement these defaults differently - or require all scalars to define this if a default is not well defined.

For Javascript, it's natural to work with Javascript objects, numbers, strings, booleans as the primary domain of data.